### PR TITLE
refactor: rename vim.highlight => vim.hl

### PR DESCRIPTION
**Pull request by justinmk**
Problem:
- `vim.highlight` module does not follow `:help dev-name-common`, which documents the name for "highlight" as "hl".
- Shorter names are usually preferred.

Solution:
Rename `vim.highlight` to `vim.hl`.

This is not a breaking change until 2.0 (or maybe never).

### CODE REVIEW

**Review by lewis6991 — APPROVED** [review]

File: MAINTAIN.md [hunk 1]
```diff
@@ -79,6 +79,8 @@ When a (non-experimental) feature is slated to be removed it should:
       `v0.10.0-dev-1957+gd676746c33` then use `0.12`.
     - For Vimscript features, use `v:lua.vim.deprecate()`. Use the same version
       as described for Lua features.
+    - `vim.deprecate(…, 'x.y.z')` where major version `x` is greater than the
+      current Nvim major version, is always treated as _soft_ deprecation.
```
Comment (justinmk) [inline MAINTAIN.md:83]: cc @dundargoc this is the only policy that makes sense when `vim.deprecate(...,'1.x')` is called

File: test/functional/lua/vim_spec.lua [hunk 2]
```diff
@@ -175,13 +182,6 @@ describe('lua stdlib', function()
         )
       end)
 
-      it('plugin=nil, to be deleted in the next major version (1.0)', function()
-        eq(
-          [[foo.baz() is deprecated. Run ":checkhealth vim.deprecated" for more information]],
-          exec_lua [[ return vim.deprecate('foo.baz()', nil, '1.0') ]]
```
Comment (justinmk) [inline test/functional/lua/vim_spec.lua:181]: Not sure if this was intentional, but doesn't really make sense. Why would we soft-deprecate `vim.deprecate(...,'0.12')` but hard-deprecate `vim.deprecate(...,'1.0')` ?

File: runtime/lua/vim/shared.lua [hunk 3]
```diff
@@ -1162,6 +1162,32 @@ function vim._defer_require(root, mod)
...
+--- @private
+--- Creates a module alias/shim that lazy-loads a target module.
+---
+--- Unlike `vim.defaulttable()` this also:
+--- - implements __call
+--- - calls vim.deprecate()
+---
+--- @param old_name string Name of the deprecated module, which will be shimmed.
+--- @param new_name string Name of the new module, which will be loaded by require().
+function vim._defer_deprecated_module(old_name, new_name)
```
Comment (justinmk) [inline runtime/lua/vim/shared.lua:1174]: I rushed this, will iterate on it later. Though if anyone sees an obvious problem, LMK.

File: MAINTAIN.md (hunk 1 above)
Comment (dundargoc) [reply MAINTAIN.md:83]: I agree, at least now when we're zerover. It's possible some adjustments might need to be made after we're no longer zerover, but there's no reason to think on that now.

File: test/functional/lua/vim_spec.lua (hunk 2 above)
Comment (dundargoc) [reply test/functional/lua/vim_spec.lua:181]: [Intentional, but still sus IMO](https://github.com/neovim/neovim/pull/27027#discussion_r1455891052). I think I'm with you that 1.0 should be soft-deprecated.

**Review by mfussenegger — APPROVED** [review]
Unrelated but does this imply we should do the same for vim.treesitter -> vim.ts ?

**Comment by lewis6991** [comment]
And `vim.filetype` -> `vim.ft`?

**Comment by clason** [comment]
And `vim` -> `vm`?

**Comment by clason** [comment]
> And `vim.filetype` -> `vim.ft`?

Less typing, but I worry a bit about this making things a bit harder to read, what with `vim.fn` and `vim.fs`...

**Comment by lewis6991** [comment]
I'm not advocating. I prefer the long names.

**Pull request merged**